DataManagementSystem/Client/FileCatalogClientCLI.py

```diff
@@ -2076,7 +2074,7 @@ def do_find(self,args):
 
       listToPrint = None
       if dirsOnly:
-        listToPrint = set( "/".join(fullpath.split("/")[:-1]) for fullpath in result['Value'] )
+        listToPrint = set( self.getPath(fullpath) for fullpath in result['Value'] )
```

Comment:
This is not doing the same thing. the join/split removes the last part of the LFN to leave just the folder.
If you want to change this line replace things with os.path.dirname